### DISCUSSION TITLE
Add Support for `Code::default()`

### DISCUSSION THREAD

**Discussion by flexchar** [laravel/nova-issues#4576] [Ideas & Feature Requests]
- Laravel Version: 9.19.0
- Nova Version: 3.32.0
- PHP Version: 8.1.4
- Operating System and Version: Ubuntu 20.04
- Browser type and version: Firefox 103.0b7 (64-bit)

### Description:

I'm trying to insert default properties using `->default()` callback on the field.
```php

Code::make('Extra Data', 'extra_data')
                ->default(function ($request) {
                       dd('finally I got called');
                       return 'default value';
                })
                ->json()
                ->autoHeight()
                ->rules('json')
                ->hideFromIndex(),
```

![image](https://user-images.githubusercontent.com/9198896/178745536-34f134dc-1ff2-43a0-8384-6a05a1be87f8.png)

However the call back seems to never be called and the field is never prefilled/always null when creating resource. 

I tried with and without dd statement. As you can see, Nova returns with `value: null`. I cannot trace it down. The model class is clean except for the cast which is set to `'extra_data' => 'object',`.

Path for the 2nd image `example.com/nova-api/model-resources/creation-fields?editing=true&editMode=create&viaResource=&viaResourceId=&viaRelationship=`
![image](https://user-images.githubusercontent.com/9198896/178744757-d66106e4-e71d-4b46-aa5f-22bec33e8ce6.png)

I may be doing something very obviously stupid or it's simply a bug.

**Comment by Livijn** [comment]
I regard this more as a bug than a feature request but the bug report was closed here: https://github.com/laravel/nova-issues/issues/3736 

In the docs, it states that we can use `default()`, but for the Code field we cannot. This is not the expected behavior. It is expected by the code of course, due to how it is written, but not by the docs nor the users. @crynobone  @davidhemphill
